airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlJdbcPartitionFactory.kt

```diff
@@ -350,15 +352,30 @@ class MysqlJdbcPartitionFactory(
                     }
                     LeafAirbyteSchemaType.TIMESTAMP_WITH_TIMEZONE -> {
                         val timestampInStatePattern = "yyyy-MM-dd'T'HH:mm:ss"
+                        val formatter =
+                            DateTimeFormatterBuilder()
+                                .appendPattern(timestampInStatePattern)
+                                .optionalStart()
+                                .appendFraction(ChronoField.NANO_OF_SECOND, 1, 9, true)
+                                .optionalEnd()
+                                .optionalStart()
+                                .optionalStart()
+                                .appendLiteral(' ')
+                                .optionalEnd()
+                                .appendOffset("+HH:mm", "Z")
+                                .optionalEnd()
+                                .toFormatter()
```

Comment:
This is painful 😓

Reply:
I found chatgpt useful for generating this kind of code. Even so, it's still annoying.

Reply:
It actually gave me wrong advice here, much to my delight 
maybe humans still got a chance 🤭

Reply:
yay humans!